src/org/skypro/skyshop/search/SearchEngine.java

```diff
@@ -13,13 +15,13 @@ public void add(Searchable searchable) {
     }
 
     // Поиск по запросу
-    public LinkedList<Searchable> search(String query) {
-        LinkedList<Searchable> results = new LinkedList<>();
+    public TreeMap<String, Searchable> search(String query) {
```

Comment:
ссылки должны быть всегда интерфейсные, чтобы соблюдать полиморфизм

```suggestion
    public Map<String, Searchable> search(String query) {
```